File: tests/fixtures/sample.go
package main

import "fmt"

func greet(name string) string {
	return fmt.Sprintf("Hello, %s", name)
}

type Calculator struct {
	result int
}

func (c *Calculator) Add(a, b int) int {
	return a + b
}

func main() {
	msg := greet("world")
	fmt.Println(msg)
}

File: tests/fixtures/sample.py
"""Sample Python file for testing tree-sitter extraction."""

import os  # noqa: F401
from pathlib import Path  # noqa: F401


def greet(name: str) -> str:
    """Greet someone by name."""
    return f"Hello, {name}"


def farewell(name: str) -> str:
    """Say goodbye, reusing greet."""
    greeting = greet(name)
    return greeting.replace("Hello", "Goodbye")


class Calculator:
    """A simple calculator."""

    def add(self, a: int, b: int) -> int:
        return a + b

    def multiply(self, a: int, b: int) -> int:
        return a * b
